### PR TITLE
SCF Assignment-6 Abstract and Concrete Data Types

### PR DESCRIPTION
**Pull request by sharma-mohit-eca**
Problem 1
Implement the immutable class intSet using an array to represent a set of integers in the range 1-1000. It should support public methods like - 
isMember(int x) - check whether x is a member of the set and return a boolean value
size() - return the size of the set
isSubSet(intSet s) - check whether s is a subset of the set
getComplement()  - return the complement set, you can assume that 1..1000 is the universal set
union(intSet s1, intSet s2) - return the union of s1 and s2
You may use private helper methods. 

Problem 2
Implement the immutable class Poly using an array to represent single variable polynomial. Note that you should be storing only those terms of polynomial that have non zero coefficient. You can assume that the coefficients are integers
It should support methods like - 
evaluate(float) - return the value of the polynomial for the given value of the variable
degree() - return the degree of the polynomial
addPoly(Poly p1, Poly p2) - return the sum of the polynomials p1 and p2
multiplyPoly(Poly p1, Poly p2) - return the product of polynomials p1 and p2
You may use private helper methods. addPoly() and multiplyPoly() can make assumptions about size of the resulting polynomial to decide about the size of the array to be created.

### CODE REVIEW

File: SCF/SCFSession6/IntSet/IntSet.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+		}
+		IntSet.sortArray(unionSetInput);
+		return unionSetInput;
+	}
+
+	/**
+	 * Sort Array using Bubble Sort
+	 * 
+	 * @param sortArray
+	 * @param lengthOfArray
```
Comment (nehajain9302) [inline SCF/SCFSession6/IntSet/IntSet.java:157]: not applicable

File: SCF/SCFSession6/IntSet/IntSet.java [hunk 2]
```diff
@@ -0,0 +1,183 @@
+package GET2018.SCF.SCFSession6.IntSet;
+
+/**
+ * This Immutable class Performs various Operations on Set
+ * 
+ * @author Mohit Sharma
+ *
+ */
+public final class IntSet {
+	static final int intSetRange[] = new int[1000];
```
Comment (nehajain9302) [inline SCF/SCFSession6/IntSet/IntSet.java:10]: why static ?

File: SCF/SCFSession6/IntSet/IntSet.java [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+			for (checkMembership = 0; checkMembership < intSetRange.length; checkMembership++) {
+				if (intSetInput[i] == intSetRange[checkMembership]) {
+					break;
+				}
+			}
+			if (checkMembership == intSetRange.length) {
+				throw new SetException("Input Invalid Not in Range 1-1000");
+			}
+		}
+		IntSet.sortArray(intSetInput);
```
Comment (nehajain9302) [inline SCF/SCFSession6/IntSet/IntSet.java]: This sorting can be done before validating range block.

File: SCF/SCFSession6/IntSet/IntSet.java [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+	}
+
+	/**
+	 * Check whether x is a member of the Set
+	 * 
+	 * @param x
+	 * @return
+	 */
+	public boolean isMember(int x) {
+		for (int i = 0; i < intSetRange.length; i++) {
```
Comment (nehajain9302) [inline SCF/SCFSession6/IntSet/IntSet.java]: Shouldn't we use intSet here ?

File: SCF/SCFSession6/IntSet/IntSetTest.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+	}
+
+	@Test
+	public void isSubSettest() throws SetException {
+		assertEquals(true, set.isSubSet(new IntSet(new int[] { 2, 35, 46 })));
+	}
+
+	@Test
+	public void getComplementtest() {
+		set.getComplement();
```
Comment (nehajain9302) [inline SCF/SCFSession6/IntSet/IntSetTest.java]: Need to assert something here. size maybe.

File: SCF/SCFSession6/IntSet/IntSet.java [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+
+	/**
+	 * Return the Union of set1 and set2
+	 * 
+	 * @param set1
+	 * @param set2
+	 * @return
+	 */
+	public static int[] union(IntSet set1, IntSet set2) {
+		int unionSet[] = new int[1000];
```
Comment (nehajain9302) [inline SCF/SCFSession6/IntSet/IntSet.java:118]: I think we can do this without using 3rd set. Please optimize this logic.

File: SCF/SCFSession6/IntSet/IntSet.java [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+			}
+		}
+		return true;
+	}
+
+	/**
+	 * Return the complement set
+	 */
+	public void getComplement() {
+		int checkMembership;
```
Comment (nehajain9302) [inline SCF/SCFSession6/IntSet/IntSet.java]: This method needs optimization as well.

File: SCF/SCFSession6/Polynomial/PolyTest.java [hunk 8]
```diff
@@ -0,0 +1,73 @@
+package GET2018.SCF.SCFSession6.Polynomial;
+
+import static org.junit.Assert.*;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import GET2018.SCF.SCFSession6.Polynomial.Poly.PolynomialException;
+
+public class PolyTest {
```
Comment (nehajain9302) [inline SCF/SCFSession6/Polynomial/PolyTest.java:10]: Please add polynomial expressions of input and output in comments.

File: SCF/SCFSession6/IntSet/IntSet.java (hunk 6 above)
Comment (sharma-mohit-eca) [reply SCF/SCFSession6/IntSet/IntSet.java:118]: The size of both sets set1 and set2 are fixed so we can't add element to these.
Therefore, we have to take a extra Set for storing all the Union Elements.

**Pull request merged**